server/job/plexsync/index.ts

```diff
@@ -182,6 +231,68 @@ class JobPlexSync {
         if (matchedtmdb?.[1]) {
           tvShow = await this.tmdb.getTvShow({ tvId: Number(matchedtmdb[1]) });
         }
+      } else if (metadata.guid.match(hamaTvdbRegex)) {
+        const matched = metadata.guid.match(hamaTvdbRegex);
+        const tvdbId = matched?.[1];
+
+        if (tvdbId) {
+          tvShow = await this.tmdb.getShowByTvdbId({ tvdbId: Number(tvdbId) });
+          if (animeList.isLoaded()) {
+            await this.processHamaSpecials(plexitem, metadata, Number(tvdbId));
+          } else {
+            this.log(
+              `Hama id ${plexitem.guid} detected, but library agent is not set to Hama`,
+              'info'
+            );
+          }
+        }
+      } else if (metadata.guid.match(hamaAnidbRegex)) {
+        const matched = metadata.guid.match(hamaAnidbRegex);
+
+        if (!animeList.isLoaded()) {
+          this.log(
+            `Hama id ${plexitem.guid} detected, but library agent is not set to Hama`,
+            'info'
+          );
+        } else if (matched?.[1]) {
+          const anidbId = Number(matched[1]);
+          const result = await animeList.getFromAnidbId(anidbId);
+          if (result) {
+            // first try to lookup tvShow by tvdbid
+            if (result.tvdbId) {
+              const extResponse = await this.tmdb.getByExternalId({
+                externalId: result.tvdbId,
+                type: 'tvdb',
+              });
+              if (extResponse.tv_results[0]) {
+                tvShow = await this.tmdb.getTvShow({
+                  tvId: extResponse.tv_results[0].id,
+                });
+              }
+              await this.processHamaSpecials(plexitem, metadata, result.tvdbId);
+            }
+
+            // if getting tvShow above failed, try with tmdb/imdb for movie
+            if (tvShow == null) {
```

Comment:
Instead just check if its truthy here. I believe `tvShow` here is undefined, not null, right? And those are treated differently in javascript. This is where strict comparison is important because you can easily have truthy values you don't expect without it. Your case here would have been fine, since undefined would come back truthy here with a non-strict comparison to null, but its unpredictable code.
```suggestion
            if (!tvShow) {
```

Reply:
tvShow is declared on line 38 to be null'able.